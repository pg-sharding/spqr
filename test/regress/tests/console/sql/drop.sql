CREATE DISTRIBUTION ds1 COLUMN TYPES integer;
CREATE KEY RANGE krid1 FROM 1 ROUTE TO sh1 FOR DISTRIBUTION ds1;
CREATE KEY RANGE krid2 FROM 11 ROUTE TO sh1 FOR DISTRIBUTION ds1;

CREATE SHARDING RULE rule1 COLUMNS id FOR DISTRIBUTION ds1;
CREATE SHARDING RULE rule2 COLUMNS idx FOR DISTRIBUTION ds1;
CREATE SHARDING RULE rule3 COLUMNS idy FOR DISTRIBUTION ds1;

SHOW key_ranges;
SHOW sharding_rules;

DROP KEY RANGE ALL;
DROP SHARDING RULE wrong;
DROP SHARDING RULE ALL;

SHOW key_ranges;
SHOW sharding_rules;

DROP DISTRIBUTION ALL CASCADE;
