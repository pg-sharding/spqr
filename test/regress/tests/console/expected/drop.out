
		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

CREATE DISTRIBUTION ds1 COLUMN TYPES integer;
    add distribution    
------------------------
 distribution id -> ds1
(1 row)

CREATE KEY RANGE krid1 FROM 1 ROUTE TO sh1 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 1
(1 row)

CREATE KEY RANGE krid2 FROM 11 ROUTE TO sh1 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 11
(1 row)

SHOW key_ranges;
 Key range ID | Shard ID | Distribution ID | Lower bound 
--------------+----------+-----------------+-------------
 krid1        | sh1      | ds1             | 1
 krid2        | sh1      | ds1             | 11
(2 rows)

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

SHOW key_ranges;
 Key range ID | Shard ID | Distribution ID | Lower bound 
--------------+----------+-----------------+-------------
(0 rows)

DROP DISTRIBUTION ALL CASCADE;
   drop distribution    
------------------------
 distribution id -> ds1
(1 row)

